Engine/ScriptAnalyzer.cs

```diff
@@ -510,75 +509,70 @@ internal bool ParseProfile(object profileObject, PathIntrinsics path, IOutputWri
-                    case "excluderules":
-                        // value must be either string or collections of string or array
-                        if (value == null
-                            || !(value is string
-                                || value is IEnumerable<string>
-                                || value.GetType().IsArray))
-                        {
-                            writer.WriteError(
-                                new ErrorRecord(
-                                    new InvalidDataException(string.Format(CultureInfo.CurrentCulture, Strings.WrongValueHashTable, value, key)),
-                                    Strings.WrongConfigurationKey,
-                                    ErrorCategory.InvalidData,
-                                    profile));
-                            hasError = true;
-                            break;
-                        }
-                        List<string> values = new List<string>();
-                        if (value is string)
-                        {
-                            values.Add(value as string);
-                        }
-                        else if (value is IEnumerable<string>)
-                        {
-                            values.Union(value as IEnumerable<string>);
-                        }
-                        else if (value.GetType().IsArray)
-                        {
-                            // for array case, sometimes we won't be able to cast it directly to IEnumerable<string>
-                            foreach (var val in value as IEnumerable)
-                            {
-                                if (val is string)
-                                {
-                                    values.Add(val as string);
-                                }
-                                else
-                                {
-                                    writer.WriteError(
-                                        new ErrorRecord(
-                                            new InvalidDataException(string.Format(CultureInfo.CurrentCulture, Strings.WrongValueHashTable, val, key)),
-                                            Strings.WrongConfigurationKey,
-                                            ErrorCategory.InvalidData,
-                                            profile));
-                                    hasError = true;
-                                    break;
-                                }
-                            }
-                        }
-                        AddProfileItem(key, values, severityList, includeRuleList, excludeRuleList);
-                        settings[key] = values;
-                        break;
+                object value = settings[settingKey];
 
-                    case "rules":
-                        break;
-
-                    default:
+                if (settingKey.Equals("severity", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
I was going to suggest changes here, but I keep thinking about the rest of the file so I might open a PR instead

Reply:
I'm going to defer this. The code works for now, but I'd really like to work on boiling down some common functionality eventually -- something we can work out in the future.